connaisseur/flask_application.py

```diff
@@ -192,6 +192,18 @@ async def __validate_image(
         )
         return
 
+    if (
+        admission_request.operation.upper() == "UPDATE"
+        and admission_request.kind == "ReplicaSet"
+        and admission_request.wl_object._spec.get("replicas", 1) == 0
```

Comment:
Didn't have a look into the k8s details, but would it be possible to issue an `UPDATE` request for a replicaset and update both the number of replicas to 0 and change the image in question? If so, this change alone in combination with automatic-allow-unchanged-approval would lead to an issue as I could set an existing replica to 0, skipping validation but changing image and then upping replica to 1, skipping validation due to unchanged approval

Reply:
yes, added additional logic to automatic update approval

Reply:
@phbelitz Sorry, I don't agree closing this without thinking about this question. Do we have an answer? (perhaps @Starkteetje?)

Reply:
@peterthomassen I mean the only thing we care about in the end are the container images (both normal and init containers). Those could be smuggled along the way when setting the replicas to 0. Everything else, we don't care about.

Should someone have smuggled another image, this will still get validated when the replicas are set >0, with the new logic on the automatic update approval.

Reply:
OK. I wasn't sure if there couldn't be some kind of other action triggered by an update request, even when replicas=0.